ui/src/app/models/new/page.tsx

```diff
@@ -264,14 +278,18 @@ function ModelPageContent() {
       if (parts.length === 2) {
         const providerKey = parts[0];
         const modelName = parts[1];
-        const baseName = `${providerKey}-${modelName}`.toLowerCase();
+        let baseName = `${providerKey}-${modelName}`.toLowerCase();
+        const isOllama = selectedProvider?.type === "Ollama";
+        if (isOllama && modelTag && modelTag !== 'latest') {
```

Comment:
nit: create a constant outside of the component for `latest` (E.g. `const LATEST_TAG="latest"`)